cognite_toolkit/cdf_tk/load.py

```diff
@@ -495,18 +504,25 @@ def get_required_capability(cls, ToolGlobals: CDFToolConfig) -> Capability:
     def get_id(cls, item: FileMetadata) -> str:
         return item.external_id
 
-    def delete(self, items: Sequence[FileMetadata]) -> None:
-        self.client.files.delete(external_id=[item.external_id for item in items])
+    def delete(self, ids: Sequence[str]) -> None:
+        self.client.files.delete(external_id=ids)
+
+    def load_file(self, filepath: Path, ToolGlobals: CDFToolConfig) -> FileMetadata:
+        file = FileMetadata.load(load_yaml_inject_variables(filepath, ToolGlobals.environment_variables()))
+        if not Path(filepath.parent / file.name).exists():
+            raise FileNotFoundError(f"Could not find file {file.name} referenced in filepath {filepath.name}")
+        if file.data_set_id is not None:
+            file.data_set_id = ToolGlobals.verify_dataset(file.data_set_id)
+        return file
 
     def create(
         self, items: Sequence[FileMetadata], ToolGlobals: CDFToolConfig, drop: bool, filepath: Path
-    ) -> FileMetadataList:
+    ) -> FileMetadata:
         if len(items) != 1:
             raise ValueError("Files must be loaded one at a time.")
         meta = items[0]
         datafile = filepath.parent / meta.name
```

Comment:
No, not exactly :)  I assume the name declared in the yaml matches the filename on disk, hence meta.name as parsed from the file. The reason is to avoid that you get a filename of file <--> filename of yaml <--> name in yaml mismatch issue. The yaml can then actually have any name (or could potentially be a single file? maybe better?), so you need to declare the name of each file in the yaml and then the metadata.

Reply:
Good point